### DISCUSSION TITLE
is there a way to force httpx to use ipv4, instead of ipv6 ?

### DISCUSSION THREAD

**Discussion by manatlan** [encode/httpx#2664] [General] [answered]
httpx or httpcore ...

tahnks in advance

**Comment by jeffsilverm** [comment]
Did you get an answer?

**Reply by karpetrosyan** [reply, marked as answer]
Use local address attribute and bind it to ipv4 address like so:

```python
import httpx

httpx.Client(transport=httpx.HTTPTransport(local_address="0.0.0.0"))
```

**Reply by jeffsilverm** [reply]
Karpestrosyan: Thank you for your prompt response, I appreciate that.  Now that I know what to look for, I see it.

I would like to ask that something be added to the documentation, something along the lines of "This is how you force httpx to use IPv4 even if it can use IPv6 (otherwise, it will use IPv6 if it can and IPv4 if it cannot)"  That way if I search for "IPv4", it will be found.  Would it be reasonable to put this request in as a documentation bug?

**Reply by karpetrosyan** [reply]
Yes, adding such line into documentation makes perfect sense to me, thanks.

I think we can include it into #3052.
